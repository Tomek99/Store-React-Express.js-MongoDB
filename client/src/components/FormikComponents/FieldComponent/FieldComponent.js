import React from "react";
import { Field } from "formik";
import ErrMessage from "../../ErrorMessage/ErrMessage";
import styles from "./FieldComponent.module.scss";

function FieldComponent({ item }) {
  return (
    <div>
      <Field
        type={item.type}
        name={item.name}
        placeholder={item.placeholder}
        className={styles.inputText}
      />
      <ErrMessage name={item.name} />
    </div>
  );
}

export default FieldComponent;
