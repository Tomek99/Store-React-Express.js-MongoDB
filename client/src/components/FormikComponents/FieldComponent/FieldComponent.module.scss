.inputText {
  width: 100%;
  border-bottom: var(--border);

  padding: 10px 0px 10px 10px;
  background: var(--bg);
  color: #fff;

  &::placeholder {
    color: #ccc;
    opacity: 0.7;
  }
}
