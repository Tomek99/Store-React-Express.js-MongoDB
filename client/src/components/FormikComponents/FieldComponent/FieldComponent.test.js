import { render } from "@testing-library/react";
import FieldComponent from "./FieldComponent";
import { Formik } from "formik";
test("should render FieldComponent component", () => {
  const item = {
    type: "text",
    name: "firstName",
    placeholder: "first name",
  };
  const { container } = render(
    <Formik>
      <FieldComponent item={item} />
    </Formik>
  );
  expect(container).toMatchSnapshot();
});
