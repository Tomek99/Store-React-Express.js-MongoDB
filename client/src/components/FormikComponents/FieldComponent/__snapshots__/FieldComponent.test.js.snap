// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FieldComponent component 1`] = `
<div>
  <div>
    <input
      class="inputText"
      name="firstName"
      placeholder="first name"
      type="text"
      value=""
    />
    <div
      class="errorText"
    />
  </div>
</div>
`;
